core/shared/src/main/scala/scalaz/zio/ZManaged.scala

```diff
@@ -73,6 +73,51 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
       }
     }
 
+  final def foldM[R1 <: R, E2, B](
+    failure: E => ZManaged[R1, E2, B],
+    success: A => ZManaged[R1, E2, B]
+  ): ZManaged[R1, E2, B] =
+    ZManaged[R1, E2, B] {
+      Ref.make[ZIO[R1, Nothing, Any]](IO.unit).map { finalizers =>
+        Reservation(
+          acquire = {
+            val direct =
+              ZIO.uninterruptibleMask { k =>
+                k(self.reserve)
+                  .flatMap(res => finalizers.update(fs => res.release *> fs).const(res))
+                  .flatMap(res => k(res.acquire))
+              }
+            val onFailure = (e: E) =>
+              ZIO.uninterruptibleMask { k =>
```

Comment:
By convention, I'd call `k` by the name `restore`, because it restores the interruptibility status of whatever the parent has.